### DISCUSSION TITLE
Can't connect to mqtt.flespi.io: Connect failed to mqtt.flespi.io:1883, rc -2. Retry in 50 sec

### DISCUSSION THREAD

**Discussion by schweini** [arendst/Tasmota#21914] [Q&A]
I just found the free MQTT tier offered by flespi.io, and would like to use it.

However, I only get the error message "`Connect failed to mqtt.flespi.io:1883, rc -2. Retry in X sec`".

As far as I can see, I have Tasmota configured correctly: I am using the loooooong string that flesp.io gives me as a username.

I have tried the same using mosquitto_pub on Linux, even forcing the use of a non-TLS connection, and it work fine there:

 `mosquitto_pub -L mqtt://XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX@mqtt.flespi.io:1883/my_topic -m "hello world"
 `
 
 I even ran a WebQuery to the same host, and that also works, so it's not a network issue:
```
 18:27:04.374 CMD: WebQuery http://mqtt.flespi.io GET
18:27:04.840 MQT: stat/tasmota_casita/RESULT = {"WebQuery":"Done"}
```

I have also tried forcing mosquitto_pub to MQTT protocol version 3.1 with -V 31, and that also works.

**Comment by s-hadinger** [comment]
Error code -2 means that the server is unreachable. Please set log to max `weblog 4` and try again. Also post all the log lines, including the ones related to dns

**Reply by s-hadinger** [reply]
Also remove everything before ‘@‘ and use user/password instead 

Tasmota does not support Mqtt links including password, and I suspect this generates a wrong dns request

**Reply by schweini** [reply]
The username@host syntax was just for testing with mosquitto_pub on Linux. In Tasmota, I used the seperate fields.

**Reply by schweini** [reply]
I set `weblog 4` and now I get
```
00:42:34.914 WIF: DNS resolved 'mqtt.flespi.io' (185.213.2.121) in 0 ms
00:42:35.136 MQT: Connect failed to mqtt.flespi.io:1883, rc -2. Retry in 20 sec
```

The device CAN connect to another MQTT server of mine on port 1883, so it doesn't seem to be some blockage on my ISP's side.

**Reply by barbudor** [reply]
When using flespi, you don't have a user/password
You have to create a token in flespi management panel and use it as user without password

I never used the unsecure option on port 1883
Always used the TLS mode on 8883 (you need a firmware built with TLS feature)
